mod-record-specifications-server/src/main/java/org/folio/rspec/service/FieldIndicatorService.java

```diff
@@ -0,0 +1,68 @@
+package org.folio.rspec.service;
+
+import java.util.UUID;
+import java.util.function.Function;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.rspec.domain.dto.FieldIndicatorChangeDto;
+import org.folio.rspec.domain.dto.FieldIndicatorDto;
+import org.folio.rspec.domain.dto.FieldIndicatorDtoCollection;
+import org.folio.rspec.domain.dto.IndicatorCodeChangeDto;
+import org.folio.rspec.domain.dto.IndicatorCodeDto;
+import org.folio.rspec.domain.dto.IndicatorCodeDtoCollection;
+import org.folio.rspec.domain.entity.Field;
+import org.folio.rspec.domain.entity.Indicator;
+import org.folio.rspec.domain.repository.IndicatorRepository;
+import org.folio.rspec.exception.ResourceNotFoundException;
+import org.folio.rspec.service.mapper.FieldIndicatorMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class FieldIndicatorService {
+
+  private final IndicatorRepository repository;
+  private final FieldIndicatorMapper mapper;
+  private final IndicatorCodeService codeService;
+
+  public FieldIndicatorDtoCollection findFieldIndicators(UUID fieldId) {
+    log.debug("findFieldIndicators::fieldId={}", fieldId);
+    var indicatorDtos = repository.findByFieldId(fieldId).stream()
+      .map(mapper::toDto)
+      .toList();
+    return new FieldIndicatorDtoCollection()
+      .indicators(indicatorDtos)
+      .totalRecords(indicatorDtos.size());
+  }
+
+  public FieldIndicatorDto createLocalIndicator(Field field, FieldIndicatorChangeDto createDto) {
+    log.info("createLocalIndicator::fieldId={}, dto={}", field.getId(), createDto);
+    var indicatorEntity = mapper.toEntity(createDto);
+    indicatorEntity.setField(field);
+    return mapper.toDto(repository.save(indicatorEntity));
+  }
+
+  @Transactional
```

Comment:
seems like a no need for doing this get operation in transaction

Reply:
We have 2 repository calls here. 1'st to get indicator and if it exists - query codes. We have the same for fields